File: src/main/scala/net/creasource/webflix/MovieExt.scala
package net.creasource.webflix

import net.creasource.json.JsonSupport
import net.creasource.tmdb.{MovieCredits, MovieDetails}
import spray.json.RootJsonFormat

case class MovieExt(
    id: Int,
    title: String,
    budget: Int,
    genres: List[MovieDetails.Genre],
    revenue: Int,
    runtime: Option[Int],
    tagline: Option[String],
    credits: MovieCredits)

object MovieExt extends JsonSupport {

  implicit val format: RootJsonFormat[MovieExt] = jsonFormat8(MovieExt.apply)

}

File: web/src/app/models/movie-ext.ts

export interface Genre {
  id: number;
  name: string;
}

export interface Cast {
  cast_id: number;
  character: string;
  credit_id: string;
  gender?: number;
  id: number;
  name: string;
  order: number;
  profile_path?: string;
}

export interface Crew {
  credit_id: string;
  department: string;
  gender?: number;
  id: number;
  job: string;
  name: string;
  profile_path?: string;
}

export interface MovieExt {
  id: number;
  title: string;
  budget: number;
  genres: Genre[];
  revenue: number;
  runtime?: number;
  tagline?: string;
  credits: {
    cast: Cast[];
    crew: Crew[];
  };
}

